torch/csrc/api/include/torch/nn/cloneable.h

```diff
@@ -21,7 +21,7 @@ namespace nn {
 /// because then storing a module would always require templatizing it.
 template <typename Derived>
 // NOLINTNEXTLINE(bugprone-exception-escape)
-class Cloneable : public virtual Module {
```

Comment:
Why make it no longer a virtual base?

Reply:
g++8 said virtual heritage combined with std::move will move virtual base object multiple times. And again , "Module" has only a string data member, there is no large benefits making it virtual. Finally, I suspect that there is no diamond object in the code base, so that virtual heritage doesn't happen.